### DISCUSSION TITLE
Question: What is the difference between Form.Field.field_name and Form.values.field_name?

### DISCUSSION THREAD

**Discussion by gmentzik** [Budibase/budibase#8930] [Help] [answered]
I want to perform checks on the field values of a form and more specifically enable a button if value is changed compared to initial from provider.
I see there are 2 ways to get the current value of the field. One by using the direct HandleBar. eg. {{ VMForm.Fields.servers }} and the other is through value object eg. {{ VMForm.Value.servers }}.
Is there any difference between these two?
Where would form value object be useful?

**Comment by melohagan** [comment, marked as answer]
Hey @gmentzik 

Good question! 

The main difference that I can see is that `{{ VMForm.Fields }}` will return nothing if you do not specify a specific value, e.g. `{{ VMForm.Fields.MyField }}`

`{{ VMForm.Value }}` will include all values of the form instance, including all fields from the schema even if there is no form field.

For example:
<img width="649" alt="Screenshot 2022-12-07 at 17 18 07" src="https://user-images.githubusercontent.com/101575380/206246713-989f1c21-cd62-4618-8e3e-37810aba9c59.png">

You can see here that the value binding includes the **_id** field. This is not available via `{{ VMForm.Fields._id }}` because it does not exist on the form, however it exists on the form schema, therefore it has a value assigned.

<img width="231" alt="Screenshot 2022-12-07 at 17 19 32" src="https://user-images.githubusercontent.com/101575380/206246992-afa4d54f-30f8-476e-9b8c-71aced444ad5.png">

The `{{ VMForm.Value }}` is what gets posted when you do a **Save Row** action. You could also store the object into [App state](https://docs.budibase.com/docs/app-state) if you wanted to save temporary *in-progress* forms.

**Reply by gmentzik** [reply]
@melohagan 
Thank you for the detailed answer and the great hint at the end!
